crates/sui-core/src/storage.rs

```diff
@@ -25,15 +24,16 @@ use sui_types::storage::WriteStore;
 use sui_types::storage::{ObjectKey, ReadStore};
 use sui_types::transaction::VerifiedTransaction;
 
-use crate::authority::AuthorityStore;
 use crate::checkpoints::CheckpointStore;
 use crate::epoch::committee_store::CommitteeStore;
 use crate::execution_cache::ExecutionCacheRead;
+use crate::execution_cache::StateSyncAPI;
 
 #[derive(Clone)]
 pub struct RocksDbStore {
-    authority_store: Arc<AuthorityStore>,
     execution_cache: Arc<dyn ExecutionCacheRead>,
+    object_store: Arc<dyn ObjectStore + Send + Sync>,
+    state_sync_store: Arc<dyn StateSyncAPI>,
```

Comment:
Please add comment note that these 3 are and must be using the same store pointer, just different trait shape because of Rust's type system limitations